File: GROKKING/selectionSort.py
def smallest(arr):
    l = 0
    for i in range(len(arr)):
        if arr[l] > arr[i]:
            l = i
    return l


def selectionSort(arr):
    newArr = []
    for i in range(len(arr)):
        small = smallest(arr)
        newArr.append(arr.pop(small))
    return newArr


nums = [2, 5, 7, 23, 4, 57, 8, 9, 127, 6, 56, 5, 4, 3, 21, 98]
print(selectionSort(nums))

File: GROKKING/selectionSort.js
// selectionsort. O(N*N
function selectionSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let low = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[low] < arr[j]) {
        low = j;
      }
    }
    let temp = arr[i];
    arr[i] = arr[low];
    arr[low] = temp;
  }
  return arr;
}



let numbers = [5, 3, 48, 16, 16, 32, 70, 44, 2, 3, 56, 49, 11];
// let numbers = [21, 3, 25, 76, 13, 45, 17, 35, 2];

console.log(selectionSort(numbers));